docs/bosh/README.md

```diff
@@ -35,42 +35,72 @@ provides an overview of the deployment:
 
 1. Configure a few environment variables:
 
-  ```
-  export project_id=$(gcloud config get-value project)
-  export region=us-east1
-  export zone=us-east1-d
-  export base_ip=10.0.0.0
-  export service_account_email=terraform@${project_id}.iam.gserviceaccount.com
-  ```
+   ```
+   export project_id=$(gcloud config get-value project)
+   export region=us-east1
+   export zone=us-east1-d
+   export base_ip=10.0.0.0
+   export service_account_email=terraform@${project_id}.iam.gserviceaccount.com
+   ```
 
 1. Configure `gcloud` to use your preferred region and zone:
 
-  ```
-  gcloud config set compute/zone ${zone}
-  gcloud config set compute/region ${region}
-  ```
+   ```
+   gcloud config set compute/zone ${zone}
+   gcloud config set compute/region ${region}
+   ```
 
 1. Create a service account and key:
 
-  ```
-  gcloud iam service-accounts create terraform
-  gcloud iam service-accounts keys create ~/terraform.key.json \
-      --iam-account ${service_account_email}
-  ```
+   ```
+   gcloud iam service-accounts create terraform
+   gcloud iam service-accounts keys create ~/terraform.key.json \
+       --iam-account ${service_account_email}
+   ```
 
 1. Grant the new service account editor access to your project:
 
-  ```
-  gcloud projects add-iam-policy-binding ${project_id} \
-    --member serviceAccount:${service_account_email} \
-    --role roles/owner
-  ```
+   ```
+   gcloud projects add-iam-policy-binding ${project_id} \
+     --member serviceAccount:${service_account_email} \
+     --role roles/owner
+   ```
+
+<a name="deploy-xpn"></a>
+### Optional: Setup Cross Project Networking (XPN)
+
+   [Cross Project Networking](https://cloud.google.com/compute/docs/xpn/_) uses a host project to manage the network resources and client project(s) to deploy compute resources. An [organization](https://cloud.google.com/resource-manager/docs/quickstart-organizations) is required to use XPN and you must be signed in as an organization admin.
+
+   The host project must have the [GCE API](https://console.developers.google.com/apis/api/compute_component/overview), [IAM API](https://console.cloud.google.com/apis/api/iam.googleapis.com/overview), and the [Cloud Resource Manager API](https://console.cloud.google.com/apis/api/cloudresourcemanager.googleapis.com/overview) enabled. This project will be used to create the bosh network throughout this guide.
+
+1. Modify and export the host project ID:
+
+   ```
+   export xpn_host_project_id=<existing project that will become the XPN host>
```

Comment:
Your comment on `${xpn_host_project_id-project_id}` is where the magic happens. If the user does not perform set this step here it will default to `project_id`. 

We can refer to the variable throughout the doc because it will be set as part of the [startup script](https://github.com/cloudfoundry-incubator/bosh-google-cpi-release/pull/179/files/ba73393df86737f035c34eee9500a0d843c96005#diff-7a438a40d1b197768537cb7ccca545a1R24)